### PR TITLE
Db Fixes #309

### LINKED ISSUE #309: Write function to run overlap SQL queries
It's really annoying having to run these by hand, we should just have functions that can run them.

### PR DESCRIPTION
**Pull request by jlaura**
Fixes #309

Tested locally on the 'deleted' db since we do not have good test coverage for executing these items via the DB.

### CODE REVIEW

File: autocnet/graph/network.py [hunk 1]
```diff
@@ -1297,11 +1298,14 @@ def __init__(self, *args, **kwargs):
         for s, d, e in self.edges(data='data'):
             e.parent = self
 
+        # Execute the computation to compute overlapping geometries
+        self._execute_sql(compute_overlaps_sql)
```
Comment (jlaura) [inline autocnet/graph/network.py]: A generic SQL submission func now exists on the NCG.

File: autocnet/graph/network.py [hunk 2]
```diff
@@ -1320,6 +1324,18 @@ def empty_queues(self):
...
+        """
+        Execute a raw SQL string in the database currently specified
+        by the AutoCNet config file.
+
+        Use this method with caution as you can easily do things like
+        truncate a table.
+        """
+        conn = engine.connect()
+        conn.execute(sql)
+        conn.close()
```
Comment (jlaura) [inline autocnet/graph/network.py]: Pretty boring here. I need to update the doc string with parameters.

File: autocnet/graph/network.py [hunk 3]
```diff
@@ -1421,10 +1437,10 @@ def generate_vrts(self, **kwargs):
         For the nodes in the graph, genreate a GDAL compliant vrt file.
         This is just a dispatcher to the knoten generate_vrt file.
         """
-        for i, n in self.nodes(data='data'):
+        for _, n in self.nodes(data='data'):
```
Comment (jlaura) [inline autocnet/graph/network.py]: Just cleanup for an unused variable in the linter.

File: autocnet/graph/network.py [hunk 4]
```diff
@@ -1459,6 +1475,23 @@ def to_isis(self, path, flistpath=None,         sql = """
         cnet.to_isis(path, df, self.serials())
         cnet.write_filelist(self.files, path=flistpath)
 
+    @classmethod
```
Comment (jlaura) [inline autocnet/graph/network.py]: This is an NCG from_filelist implementation. The initial if/else block is not great. The two other ways I see to implemented this are (1) using multiple dispatch on the file list or (2) using a try/except. It felt like this version was the most readable. If the filets is an utterable, just pass. Else pass to plio to open the file.

File: autocnet/graph/network.py [hunk 5]
```diff
@@ -1515,6 +1548,6 @@ def from_database(cls, query_string='SELECT * FROM public.images'):
                 adjacency[spath].append(dpath)
         session.close()
         # Add nodes that do not overlap any images
-        obj = cls.from_adjacency(adjacency, node_id_map=adjacency_lookup, config=config)
+        obj = cls(adjacency, node_id_map=adjacency_lookup, config=config)
```
Comment (jlaura) [inline autocnet/graph/network.py]: As I dug through the networkX docs I could not find any reference to `from_adjacency`. Turns out the default constructor is `from_adjacency` so all good there.

File: autocnet/spatial/overlap.py [hunk 6]
```diff
@@ -103,3 +103,22 @@ def place_points_in_overlaps(cg, size_threshold=0.0007, reference=None, height=0
...
+ SELECT images.id, intersectiongeom.geom AS geom
+		FROM images, intersectiongeom
+		WHERE images.footprint_latlon is NOT NULL AND
+		ST_INTERSECTS(intersectiongeom.geom, images.footprint_latlon) AND
+		ST_AREA(ST_INTERSECTION(intersectiongeom.geom, images.footprint_latlon)) > 0.000001
+)
+INSERT INTO overlay(intersections, geom) SELECT row.intersections, row.geom FROM 
+(SELECT iid.geom, array_agg(iid.id) AS intersections
+  FROM iid GROUP BY iid.geom) AS row WHERE array_length(intersections, 1) > 1;
+"""
```
Comment (jlaura) [inline autocnet/spatial/overlap.py]: ... 👎 , but I still get a 🏅 for coming up with whatever this is.

**Comment by jlaura** [comment]
@all I restarted the build and all passing. Failures were on the conda side installing the dependencies. Looks like a network issue was to blame.

File: autocnet/graph/network.py [hunk 7]
```diff
@@ -1459,6 +1475,23 @@ def to_isis(self, path, flistpath=None,         sql = """
...
+        elif os.path.exists(filelist):
+            filelist = io_utils.file_to_list(filelist)
+        else:
+            warning.warn('Unable to parse the passed filelist')
+
+        for f in filelist:
+            # Create the nodes in the graph. Really, this is creating the
+            # images in the DB
+            image_name = os.path.basename(f)
+            NetworkNode(image_path=f, image_name=image_name)
```
Comment (Kelvinrr) [inline autocnet/graph/network.py]: NetworkNode init handles footprint generation and overlap computation?

File: autocnet/graph/network.py [hunk 8]
```diff
@@ -1459,6 +1475,23 @@ def to_isis(self, path, flistpath=None,         sql = """
...
+        
+        return cls.from_database()
```
Comment (Kelvinrr) [inline autocnet/graph/network.py]: 👍

File: autocnet/spatial/overlap.py (hunk 6 above)
Comment (Kelvinrr) [reply autocnet/spatial/overlap.py]: Any way to use the query from the sql folder instead of duplicating?

File: autocnet/graph/network.py (hunk 7 above)
Comment (jlaura) [reply autocnet/graph/network.py]: yes, the properties are lazy loaded and if they do not exist they are computed

File: autocnet/spatial/overlap.py (hunk 6 above)
Comment (jlaura) [reply autocnet/spatial/overlap.py]: I should remove those queries. This one supersedes that one and it is cleaner to have in one place. Any issue with me removing those?

Comment (Kelvinrr) [reply autocnet/spatial/overlap.py]: nah just remove them

Comment (jlaura) [reply autocnet/spatial/overlap.py]: Done.

**Comment by jlaura** [comment]
@Kelvinrr I think that this should be good to go. I do not see CI processing again on the deleted file (which is maybe a little concerning...not that the sql was being tested).

**Pull request merged**